modules/ting_search_ortograf/ting_search_ortograf.module

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * @file
+ * Add auto-complete functions to the ting search field, using the Ortograf service.
+ */
+
+/**
+ * Implements hook_menu().
+ */
+function ting_search_ortograf_menu() {
+  $items = array();
+
+  $items['ting/autocomplete'] = array(
+    'title' => 'Ting search autocomplete',
+    'description' => 'Returns a list of suggestions to use for autocompletion',
+    'access arguments' => array('search content'),
+    'page callback' => 'ting_search_ortograf_callback',
+    'file' => 'includes/ting_search_ortograf.ajax.inc',
+    'type' => MENU_CALLBACK,
+  );
+
+  $items['admin/config/ting/ting_search_ortograf'] = array(
+    'title' => 'Ting search autocomplete settings',
+    'description' => 'Settings for Ting search autocomplete, using the Ortograf service',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ting_search_ortograf_admin'),
+    'access arguments' => array('administer search'),
+    'file' => 'includes/ting_search_ortograf.admin.inc',
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_flush_caches().
+ */
+function ting_search_ortograf_flush_caches() {
+  return array('cache_ting_search_ortograf');
+}
+
+/**
+ * Implements hook_form_alter().
+ *
+ * Adds CSS and JS to the ting search form.
+ */
+function ting_search_ortograf_form_search_block_form_alter(&$form, &$form_state) {
+  $form['search_block_form']['#autocomplete_path'] = 'ting/autocomplete';
+
+  // Add class to make sure that the form is auto-submitted.
+  $form['search_block_form']['#attributes']['class'] = array('auto-submit');
+
+  // Modify the core auto-complete behaviour to search on enter. It has a high
+  // weight to ensure that it's include after core auto complete script.
+  drupal_add_js(drupal_get_path('module', 'ting_search_ortograf') . '/js/ting_search_ortograf.js', array('weight' => 1000));
+}
+
+
+/**
+ * Get settings, and merge with defaults.
+ *
+ * @return array
+ *   Search suggestion settings.
+ */
+function ting_search_ortograf_settings() {
+  $settings = array();
```

Comment:
We do not need to set `$settings` here. It is not accessed before it is overwritten on L70.